core/src/cluster_slots.rs

```diff
@@ -195,7 +195,10 @@ impl ClusterSlots {
             .iter()
             .map(|peer| slot_peers.get(&peer.id).cloned().unwrap_or(0))
             .zip(stakes)
-            .map(|(a, b)| a + b)
+            .map(|(a, b)| {
+                let non_zero = a > 0 || b > 0;
+                (a / 2 + b / 2).max(u64::from(non_zero))
```

Comment:
I think we probably want `.max(1u64)` anyways.
e.g. I am not sure how `WeightedIndex` would behave if all weights are zero.

Reply:
changed. the prior stake calculation would have ensured this would always be `1`.